### PR TITLE
Update eth-tester to Istanbul compatibility, update web3.

### LINKED ISSUE #868: Update to eth-tester for Istanbul compatibility
According to the [changelog](https://github.com/ethereum/eth-tester/blob/master/CHANGELOG) this sets the default VM rules to Constantinople

### PR DESCRIPTION
**Pull request by palango**
### What this PR does

Reincarnation of #1153 
Fixes #868

Updates web3 and eth-tester to recent versions. This lets us run the tests on an instanbul enabled test chain.

### Why I'm making this PR

We want to test the contracts with the latest ethereum versions and functionality.

### What's tricky about this PR (if any)

Nothing

----

Any reviewer can check these:

* [ ] If the PR is fixing a bug or adding a feature, add an entry to CHANGELOG.md.
* [ ] If the PR changed a Solidity source, run `make compile_contracts` and add the resulting `raiden_contracts/data/contracts.json` in the PR.
* [ ] If the PR is changing documentation only, add `[skip ci]` in the commit message so Travis does not waste time.
    * [ ] But, if the PR changes comments in a Solidity source, do not add `[skip ci]` and let Travis check the hash of the source.
* [ ] In Python, use keyword arguments
* [ ] Squash unnecessary commits
* [ ] Comment commits
* [ ] Follow naming conventions
    * `solidityFunction`
    * `_solidity_argument`
    * `solidity_variable`
    * `python_variable`
    * `PYTHON_CONSTANT`
* [ ] Follow the Signature Convention in [CONTRIBUTING.md](./CONTRIBUTING.md)
* For each new contract
    * [ ] The deployment script deploys the new contract.
    * [ ] `etherscan_verify.py` runs on the new contract.
* Bookkeep
    * [ ] The gas cost of new functions are stored in `gas.json`.
* Solidity specific conventions
    * [ ] Document arguments of functions in natspec
    * [ ] Care reentrancy problems
    * if the PR adds or removes `require()` or `assert()`
        * [ ] add an entry in Changelog
        * [ ] open an issue in the client, light client, service repos so the change is reflected there
        * Just adding a message in `require()` doesn't require these steps.
* [ ] When you catch a require() failure in Solidity, look for a specific error message like `pytest.raises(TransactionFailed, match="error message"):`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

### CODE REVIEW

File: raiden_contracts/tests/deprecation_switch_testnet.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def deprecation_test(
 
     # Check that we can open channels and deposit on behalf of A and B
     # Some arbitrary Ethereum addresses
-    A = "0x6AA63296FA94975017244769F00F0c64DB7d7115"
-    B = "0xc9a4fad99B6d7D3e48D18d2585470cd8f27FA61e"
+    A = HexAddress(HexStr("0x6AA63296FA94975017244769F00F0c64DB7d7115"))
```
Comment (karlb) [inline raiden_contracts/tests/deprecation_switch_testnet.py]: Why do we need the `HexStr`? Can't we cast to `HexAddress` directly?

File: raiden_contracts/tests/fixtures/base/web3_fixtures.py [hunk 2]
```diff
@@ -56,6 +56,9 @@ def web3(ethereum_tester: EthereumTester) -> Web3:
         }
     )
 
+    # Enable strict type checks
+    web3.enable_strict_bytes_type_checking()
```
Comment (karlb) [inline raiden_contracts/tests/fixtures/base/web3_fixtures.py]: Nice!

File: raiden_contracts/tests/deprecation_switch_testnet.py (hunk 1 above)
Comment (palango) [reply raiden_contracts/tests/deprecation_switch_testnet.py]: Yeah, that one is weird:
```
mypy raiden_contracts/ setup.py
raiden_contracts/tests/deprecation_switch_testnet.py:68: error: Argument 1 to "HexAddress" has incompatible type "str"; expected "HexStr"

Comment (karlb) [reply raiden_contracts/tests/deprecation_switch_testnet.py]: It's really defined that way:
```
HexAddress = NewType('HexAddress', HexStr)
```
see https://github.com/ethereum/eth-typing/blob/965459459f1a4f1090ccc004b433c11fb4ad97b4/eth_typing/evm.py#L16

We've never cascaded NewTypes in raiden, so this was unusual for me.

**Pull request merged**